### PR TITLE
server: Implement get public key from secret key

### LINKED ISSUE #49: Implement a function that gets the public key from the private key.
## Feature Request

**Is your feature request related to a problem? Please describe.**  

Currently, our library does not provide a function to get a public key from a private key string.

**Describe the solution you'd like**  

Implement a wrapper function to get the public key from private key. (`elliptic` library already has a function that does the same thing.)

**Describe alternatives you've considered**  

**Additional context**

### PR DESCRIPTION
**Pull request by HyeockJinKim**
# Summary 

- It closes #49

## How to Review? 

- Please refer to my self-review submitted as the review.

### CODE REVIEW

File: packages/server/src/index.ts [hunk 1]
```diff
@@ -37,13 +27,25 @@ type SubmitInfo<T> = {
   confirm: Supplier<T, T | null>;
 };
 
+export function publicKeyFromSecret(secret: SecretKey): PublicKey {
+  return {
+    key: publicKeyFromKeyString(secret.key, secret.type),
+    type: secret.type,
+  };
+}
+
```
Comment (HyeockJinKim) [inline packages/server/src/index.ts:36]: secretKey type to publicKey type function.

File: packages/server/src/utils.ts [hunk 2]
```diff
@@ -18,29 +18,38 @@ export function getMerkleRoot(properties: string[]): string {
...
+      throw new Error('Unreachable Code');
+  }
+}
+
 export function signByKeyType(
   hashValue: string,
   secret: string,
   type: KeyType,
-): Signature {
+): string {
```
Comment (HyeockJinKim) [inline packages/server/src/utils.ts:42]: `sign` function only returns string.
`auth` functions return Signature type.

File: packages/server/test/auth.spec.ts [hunk 3]
```diff
@@ -12,6 +20,16 @@ describe('test server utils', () => {
...
+  it('test public key from key string in ECDSA', () => {
+    const secretKeyString = '2ef40452ec154cd38efdc8ffa52e7f513f7d2b2a77e028342bde96c369e4f77a';
+    const publicKey = publicKeyFromKeyString(secretKeyString, KeyType.ECDSA);
+    expect(publicKey).to.be.equal('048da7b63430eb4db203177baf2e8699a25116561624e67a31c2bf288d54216ce3f6f9c7b81fdbb5732342475a6ee5ccab883277ddbb38fdb79ab5424d401b844a');
+  });
+  it('test public key from key string in EDDSA', () => {
+    const secretKeyString = '2ef40452ec154cd38efdc8ffa52e7f513f7d2b2a77e028342bde96c369e4f77a';
+    const publicKey = publicKeyFromKeyString(secretKeyString, KeyType.EDDSA);
+    expect(publicKey).to.be.equal('cb7da1efe0ca47d03ff12d1b8f01160debf62c0a2cb2517251f3019d61e0c5f3');
+  });
```
Comment (HyeockJinKim) [inline packages/server/test/auth.spec.ts:32]: test for `publicKeyFromKeyString`

File: packages/server/test/auth.spec.ts [hunk 4]
```diff
@@ -26,13 +44,14 @@ describe('test privacy mode authentication', () => {
...
-    const sign = authPrivacy(properties, { key: key1.getPrivate('hex'), type: KeyType.ECDSA });
+    const secret1 = { key: key1.getPrivate('hex'), type: KeyType.ECDSA };
+    const sign = authPrivacy(properties, secret1);
 
     expect(key1.verify(merkleRoot, sign.value)).to.be.equal(true);
     expect(key2.verify(merkleRoot, sign.value)).to.be.equal(false);
     expect(verifyByKeyType(merkleRoot, sign.value, key1.getPublic('hex'), KeyType.ECDSA)).to.be.equal(true);
     expect(verifyByKeyType(merkleRoot, sign.value, key2.getPublic('hex'), KeyType.ECDSA)).to.be.equal(false);
-    expect(verifyPrivacy(properties, sign, { key: key1.getPublic('hex'), type: KeyType.ECDSA })).to.be.equal(true);
+    expect(verifyPrivacy(properties, sign, publicKeyFromSecret(secret1))).to.be.equal(true);
```
Comment (HyeockJinKim) [inline packages/server/test/auth.spec.ts:54]: use `publicKeyFromSecret` function.

**Review by HyeockJinKim — COMMENTED** [review]
self review

**Review by anthonyminyungi — APPROVED** [review]
Looks good

**Pull request merged**